runtime/vm/ValueTypeHelpers.cpp

```diff
@@ -69,4 +69,85 @@ isClassRefQtype(J9Class *cpContextClass, U_16 cpIndex)
 {
 	return VM_ValueTypeHelpers::isClassRefQtype((J9ConstantPool *) cpContextClass->ramConstantPool, cpIndex);
 }
+
+
+
+
+/**
+ * Performs an aastore operation on an object. Handles flattened and non-flattened cases.
+ *
+ * If you can guarantee that receiverObject and paramObj will never be null then it
+ * doesnt need to throw.
+ *
+ * @param[in] currentThread thread token
+ * @param[in] receiverObject arrayobject
+ * @param[in] index array index
+ * @param[in] paramObject obj arg
+ *
+ */
+void
+jitAastore(J9VMThread currentThread, j9_object_t *arrayObject, U_32 index, j9_object_t *paramObject)
```

Comment:
this will have to be added to a function table, same as the other one. you can rename them to [load/store]FlattenableArrayElement